File: AdminApi.Sdk/Endpoints/Tiers/Types/TierDetails.cs
using Backbone.AdminApi.Sdk.Endpoints.Identities.Types;

namespace Backbone.AdminApi.Sdk.Endpoints.Tiers.Types;

public class TierDetails
{
    public required string Id { get; set; }
    public required string Name { get; set; }
    public required List<TierQuotaDefinition> Quotas { get; set; }
}

File: Applications/AdminUi/packages/admin_api_types/lib/src/tiers/tier_details.dart
import 'package:json_annotation/json_annotation.dart';

import 'tier_quota_definition.dart';

part 'tier_details.g.dart';

@JsonSerializable()
class TierDetails {
  final String id;
  final String name;
  final List<TierQuotaDefinition> quotas;

  TierDetails({required this.id, required this.name, required this.quotas});

  factory TierDetails.fromJson(dynamic json) => _$TierDetailsFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$TierDetailsToJson(this);
}
